frontend/src/components/pages/RegistrantExperience/RegistrationSteps/ReviewRegistration/EditContactCard.tsx

```diff
@@ -0,0 +1,399 @@
+        });
+
+        dispatchPersonalInfoAction({
+          type: PersonalInfoActions.UPDATE_CONTACT,
+          field: "lastName",
+          contactIndex,
+          data: initialContact.lastName,
+        });
+
+        dispatchPersonalInfoAction({
+          type: PersonalInfoActions.UPDATE_CONTACT,
+          field: "email",
+          contactIndex,
+          data: initialContact.email,
+        });
+
+        dispatchPersonalInfoAction({
+          type: PersonalInfoActions.UPDATE_CONTACT,
+          field: "phoneNumber",
+          contactIndex,
+          data: initialContact.phoneNumber,
+        });
+
+        dispatchPersonalInfoAction({
+          type: PersonalInfoActions.UPDATE_CONTACT,
+          field: "relationshipToCamper",
+          contactIndex,
+          data: initialContact.relationshipToCamper,
+        });
+
+        setEditing(false);
+
+        toast({
+          description: `Edits have been discarded.`,
+          status: "success",
+          duration: 3000,
+          isClosable: false,
+          variant: "subtle",
+        });
+        onClose();
+      },
+    });
+
+    return (
+      <>
+        <Button
+          color="text.critical.100"
+          variant="outline"
+          onClick={onOpen}
+          colorScheme="red"
+          w={{ sm: "80px", lg: "100px" }}
+          h={{ sm: "30px", lg: "40px" }}
+        >
+          <Text textStyle={{ sm: "xSmallBold", lg: "buttonSemiBold" }}>
+            Cancel
+          </Text>
+        </Button>
+        {deleteCamperModal}
+      </>
+    );
+  };
+
+  return (
+    <Box boxShadow="lg" rounded="xl" borderWidth={1} width="100%" mt={2} mb={2}>
+      <Box backgroundColor="#FFFFFF" rounded="xl">
+        <Heading textStyle="displayLarge">
+          <Flex py={6} px={{ sm: "5", lg: "20" }} alignItems="center">
+            <Text textStyle={{ sm: "xSmallBold", lg: "displayLarge" }}>
+              {contactIndex === 0 ? "Primary Contact" : "Secondary Contact"}{" "}
+            </Text>
+            <Spacer />
+            <Button
+              variant="secondary"
+              textStyle={{ sm: "xSmallBold", lg: "buttonSemiBold" }}
+              disabled={editing}
+              w={{ sm: "80px", lg: "100px" }}
+              h={{ sm: "30px", lg: "40px" }}
+              onClick={() => setEditing(true)}
+            >
+              Edit
+            </Button>
+          </Flex>
+        </Heading>
+        <Divider borderColor="border.secondary.100" />
+      </Box>
+
+      <Box
+        zIndex={0}
+        backgroundColor="#FFFFFFAA"
+        borderRadius="0px 0px 10px 10px"
+        _hover={{ cursor: editing ? "auto" : "not-allowed" }}
+      >
+        <Box zIndex={editing ? 1 : -1} position="relative">
+          <Box px={{ sm: "5", lg: "20" }}>
+            <Wrap pt="7">
+              <WrapItem width={{ sm: "100%", md: "45%", lg: "20%" }}>
+                <FormControl isInvalid={isFirstNameInvalid}>
+                  <FormLabel>
```

Comment:
If this has the same styling as the EditCamperCard, we could just use that component for the formLabel and pass in the title as well as a boolean prop to indicate if it's required or not

Reply:
Done.